clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/GroupMetadataUpdateEvent.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.consumer.internals.events;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.internals.ConsumerNetworkThread;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * This event is sent by the {@link ConsumerNetworkThread consumer's network thread} to the application thread
+ * so that when the user calls the {@link Consumer#groupMetadata()} API, the information is up-to-date. The
+ * information for the current state of the group member is managed on the consumer network thread and thus
+ * requires this interplay between threads.
+ */
+public class GroupMetadataUpdateEvent extends BackgroundEvent {
+
+    final private String groupId;
+    final private int memberEpoch;
+    final private String memberId;
+    final private Optional<String> groupInstanceId;
+
+    public GroupMetadataUpdateEvent(final String groupId,
+                                    final int memberEpoch,
+                                    final String memberId,
+                                    final Optional<String> groupInstanceId) {
+        super(Type.GROUP_METADATA_UPDATED);
```

Comment:
Sorry for being a naming fundamentalist, but you've used `GroupMetadataUpdateEvent` but `GROUP_METADATA_UPDATED`. Please use the same tense for the class and the constant. I'm sorting out the subclasses of `ApplicationEvent` in a similar vein in another PR. Just makes it slightly easier to navigate the code.